--- origsrc/texlive-20120628-source/texk/web2c/luatexdir/luazlib/lzlib.c	2009-03-27 11:30:55.000000000 -0400
+++ src/texlive-20120628-source/texk/web2c/luatexdir/luazlib/lzlib.c	2013-05-25 14:14:49.368961700 -0400
@@ -549,14 +549,6 @@ LUALIB_API int luaopen_zlib(lua_State *L
 
     /* ====================================================================== */
 
-    /* make sure header and library version are consistent */
-    const char* version = zlibVersion();
-    if (strncmp(version, ZLIB_VERSION, 5))
-    {
-        lua_pushfstring(L, "zlib library version does not match - header: %s, library: %s", ZLIB_VERSION, version);
-        lua_error(L);
-    }
-
     /* create new metatable for zlib compression structures */
     luaL_newmetatable(L, ZSTREAMMETA);
     lua_pushliteral(L, "__index");
