--- origsrc/tlpkg/tlpobj/texlive-scripts.i386-cygwin.tlpobj	2013-04-10 21:44:33.000000000 -0400
+++ src/tlpkg/tlpobj/texlive-scripts.i386-cygwin.tlpobj	2014-03-31 14:46:32.475444800 -0400
@@ -3,5 +3,5 @@ category TLCore
 revision 29836
 shortdesc i386-cygwin files of texlive-scripts
 binfiles arch=i386-cygwin size=2
- bin/i386-cygwin/man
+# bin/i386-cygwin/man
  bin/i386-cygwin/rungs
--- origsrc/tlpkg/tlpobj/texlive-scripts.tlpobj	2014-03-30 21:22:31.000000000 -0400
+++ src/tlpkg/tlpobj/texlive-scripts.tlpobj	2014-03-31 14:48:50.474337900 -0400
@@ -8,15 +8,15 @@ depend texlive.infra
 depend texlive-scripts.ARCH
 postaction shortcut type=menu name="TeX Live command-line"   cmd=TEXDIR/tlpkg/installer/tl-cmd.bat
 runfiles size=65
- install-tl
- texmf-dist/scripts/texlive/lua/texlive/getopt.tlu
- texmf-dist/scripts/texlive/lua/texlive/tlpdb.tlu
- texmf-dist/scripts/texlive/lua/texlive/utils.tlu
+# install-tl
+# texmf-dist/scripts/texlive/lua/texlive/getopt.tlu
+# texmf-dist/scripts/texlive/lua/texlive/tlpdb.tlu
+# texmf-dist/scripts/texlive/lua/texlive/utils.tlu
  texmf-dist/scripts/texlive/rungs.tlu
- texmf-dist/scripts/texlive/test-tlpdb.tlu
- texmf-dist/scripts/texlive/texconf.tlu
- tlpkg/installer/ctan-mirrors.pl
- tlpkg/installer/install-menu-perltk.pl
- tlpkg/installer/install-menu-text.pl
- tlpkg/installer/install-menu-wizard.pl
- tlpkg/installer/texlive.png
+# texmf-dist/scripts/texlive/test-tlpdb.tlu
+# texmf-dist/scripts/texlive/texconf.tlu
+# tlpkg/installer/ctan-mirrors.pl
+# tlpkg/installer/install-menu-perltk.pl
+# tlpkg/installer/install-menu-text.pl
+# tlpkg/installer/install-menu-wizard.pl
+# tlpkg/installer/texlive.png
