--- origsrc/tlpkg/tlpobj/luatex.tlpobj	2014-05-04 18:20:52.000000000 -0400
+++ src/tlpkg/tlpobj/luatex.tlpobj	2014-06-24 17:12:15.840726900 -0400
@@ -14,7 +14,7 @@ longdesc absolute stability may not in p
 depend luatex.ARCH
 execute AddFormat   name=luatex    engine=luatex patterns=language.def,language.dat.lua   options="luatex.ini"
 execute AddFormat   name=dviluatex engine=luatex patterns=language.def,language.dat.lua   options="dviluatex.ini"
-execute AddFormat   name=luajittex engine=luajittex patterns=language.def,language.dat.lua   options="luatex.ini"
+execute not AddFormat   name=luajittex engine=luajittex patterns=language.def,language.dat.lua   options="luatex.ini"
 runfiles size=37
  texmf-dist/tex/generic/config/luatex-unicode-letters.tex
  texmf-dist/tex/generic/config/luatexiniconfig.tex
