--- origsrc/bin/x86_64-cygwin/mktexlsr	2015-04-14 08:17:33.000000000 -0400
+++ src/bin/x86_64-cygwin/mktexlsr	2015-06-13 09:24:45.436570700 -0400
@@ -73,7 +73,7 @@ if tty -s; then verbose=true; else verbo
 dry_run=false
 trees=
 
-treefile="${TMPDIR-/tmp}/mktexlsrtrees$$.tmp"
+treefile=`mktemp -q --tmpdir mktexlsrtrees.XXXXXXXXXX`
 trap 'cd /; rm -f $treefile; test -z "$db_dir_tmp" || rm -rf "$db_dir_tmp"; 
       exit' 0 1 2 3 7 13 15
 
