--- origsrc/texmf-dist/web2c/texmf.cnf	2015-10-01 11:39:06.646577000 -0400
+++ src/texmf-dist/web2c/texmf.cnf	2015-10-01 11:42:48.231874300 -0400
@@ -229,7 +229,7 @@ TEXINPUTS.platex-ng = .;$TEXMF/tex/{upla
 % (e)up(La)TeX, and for upmpost
 TEXINPUTS.uplatex = .;$TEXMF/tex/{uplatex,platex,latex,generic,}//
 TEXINPUTS.uptex   = .;$TEXMF/tex/{uptex,ptex,plain,generic,}//
-TEXINPUTS.euptex  = .;$TEXMF/tex/{uptex,ptex,plain,generic}//
+TEXINPUTS.euptex  = .;$TEXMF/tex/{uptex,ptex,plain,generic,}//
 TEX.upmpost = euptex
 
 % pBibTeX bibliographies and style files.
