--- origsrc/texlive-20150521-source/libs/luajit/LuaJIT-2.0.3/src/lj_alloc.c	2014-03-14 05:24:13.000000000 -0400
+++ src/texlive-20150521-source/libs/luajit/LuaJIT-2.0.3/src/lj_alloc.c	2015-09-20 15:40:01.437204700 -0400
@@ -188,7 +188,7 @@ static LJ_AINLINE void *CALL_MMAP(size_t
   return ptr;
 }
 
-#elif LJ_TARGET_OSX || LJ_TARGET_PS4 || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__sun__)
+#elif LJ_TARGET_OSX || LJ_TARGET_PS4 || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__sun__) || defined(__CYGWIN__)
 
 /* OSX and FreeBSD mmap() use a naive first-fit linear search.
 ** That's perfect for us. Except that -pagezero_size must be set for OSX,
--- origsrc/texlive-20150521-source/libs/luajit/LuaJIT-2.0.3/src/lj_arch.h	2015-02-19 07:12:35.000000000 -0500
+++ src/texlive-20150521-source/libs/luajit/LuaJIT-2.0.3/src/lj_arch.h	2015-09-20 15:38:11.180342500 -0400
@@ -145,7 +145,11 @@
 #define LJ_ARCH_NAME		"x64"
 #define LJ_ARCH_BITS		64
 #define LJ_ARCH_ENDIAN		LUAJIT_LE
-#define LJ_ABI_WIN		LJ_TARGET_WINDOWS
+#if LJ_TARGET_WINDOWS || __CYGWIN__
+#define LJ_ABI_WIN		1
+#else
+#define LJ_ABI_WIN		0
+#endif
 #define LJ_TARGET_X64		1
 #define LJ_TARGET_X86ORX64	1
 #define LJ_TARGET_EHRETREG	0
--- origsrc/texlive-20150521-source/libs/luajit/LuaJIT-2.0.3/src/lj_err.c	2014-03-14 05:24:13.000000000 -0400
+++ src/texlive-20150521-source/libs/luajit/LuaJIT-2.0.3/src/lj_err.c	2015-09-20 15:38:11.180342500 -0400
@@ -186,7 +186,7 @@ static void *err_unwind(lua_State *L, vo
 
 /* -- External frame unwinding -------------------------------------------- */
 
-#if defined(__GNUC__) && !LJ_NO_UNWIND && !LJ_TARGET_WINDOWS
+#if defined(__GNUC__) && !LJ_NO_UNWIND && !LJ_ABI_WIN
 
 /*
 ** We have to use our own definitions instead of the mandatory (!) unwind.h,
@@ -352,7 +352,7 @@ LJ_FUNCA int lj_err_unwind_arm(int state
 
 #endif
 
-#elif LJ_TARGET_X64 && LJ_TARGET_WINDOWS
+#elif LJ_TARGET_X64 && LJ_ABI_WIN
 
 /*
 ** Someone in Redmond owes me several days of my life. A lot of this is
@@ -417,7 +417,9 @@ LJ_FUNCA EXCEPTION_DISPOSITION lj_err_un
     if (cf2) {  /* We catch it, so start unwinding the upper frames. */
       if (rec->ExceptionCode == LJ_MSVC_EXCODE ||
 	  rec->ExceptionCode == LJ_GCC_EXCODE) {
+#if LJ_TARGET_WINDOWS
 	__DestructExceptionObject(rec, 1);
+#endif
 	setstrV(L, L->top++, lj_err_str(L, LJ_ERR_ERRCPP));
       } else if (!LJ_EXCODE_CHECK(rec->ExceptionCode)) {
 	/* Don't catch access violations etc. */
--- origsrc/texlive-20150521-source/m4/kpse-setup.m4	2015-01-14 07:19:37.000000000 -0500
+++ src/texlive-20150521-source/m4/kpse-setup.m4	2015-09-20 15:45:49.651662200 -0400
@@ -84,7 +84,7 @@ AS_CASE([$with_x:$kpse_cv_have_win32],
 AS_CASE([$enable_luajittex],
         [yes | no], [:],
           [AS_CASE([$host],
-                   [alpha* | sparc* | x86_64-*-cygwin | powerpc-*-darwin* ],
+                   [alpha* | sparc* | powerpc-*-darwin* ],
                      [AC_MSG_NOTICE([$host -> `--disable-luajittex'])
                       ac_configure_args="$ac_configure_args '--disable-luajittex'"])])        
 KPSE_FOR_PKGS([utils], [m4_sinclude(kpse_TL[utils/]Kpse_Pkg[/ac/withenable.ac])])
