--- origsrc/tlpkg/tlpobj/texlive-scripts.tlpobj	2015-05-03 21:21:38.000000000 -0400
+++ src/tlpkg/tlpobj/texlive-scripts.tlpobj	2015-05-05 15:00:04.922831200 -0400
@@ -8,15 +8,15 @@ depend texlive.infra
 depend texlive-scripts.ARCH
 postaction shortcut type=menu name="TeX Live command-line"   cmd=TEXDIR/tlpkg/installer/tl-cmd.bat
 runfiles size=66
- install-tl
- texmf-dist/scripts/texlive/lua/texlive/getopt.tlu
- texmf-dist/scripts/texlive/lua/texlive/tlpdb.tlu
- texmf-dist/scripts/texlive/lua/texlive/utils.tlu
+# install-tl
+# texmf-dist/scripts/texlive/lua/texlive/getopt.tlu
+# texmf-dist/scripts/texlive/lua/texlive/tlpdb.tlu
+# texmf-dist/scripts/texlive/lua/texlive/utils.tlu
  texmf-dist/scripts/texlive/rungs.tlu
- texmf-dist/scripts/texlive/test-tlpdb.tlu
- texmf-dist/scripts/texlive/texconf.tlu
- tlpkg/installer/ctan-mirrors.pl
- tlpkg/installer/install-menu-perltk.pl
- tlpkg/installer/install-menu-text.pl
- tlpkg/installer/install-menu-wizard.pl
- tlpkg/installer/texlive.png
+# texmf-dist/scripts/texlive/test-tlpdb.tlu
+# texmf-dist/scripts/texlive/texconf.tlu
+# tlpkg/installer/ctan-mirrors.pl
+# tlpkg/installer/install-menu-perltk.pl
+# tlpkg/installer/install-menu-text.pl
+# tlpkg/installer/install-menu-wizard.pl
+# tlpkg/installer/texlive.png
--- origsrc/tlpkg/tlpobj/texlive-scripts.x86_64-cygwin.tlpobj	2015-04-12 20:27:47.000000000 -0400
+++ src/tlpkg/tlpobj/texlive-scripts.x86_64-cygwin.tlpobj	2015-05-05 14:56:31.306449300 -0400
@@ -3,5 +3,5 @@ category TLCore
 revision 33557
 shortdesc x86_64-cygwin files of texlive-scripts
 binfiles arch=x86_64-cygwin size=2
- bin/x86_64-cygwin/man
+# bin/x86_64-cygwin/man
  bin/x86_64-cygwin/rungs
