--- origsrc/texlive-20150521-source/texk/kpathsea/texmf.cnf	2015-09-21 17:32:37.070843600 -0400
+++ src/texlive-20150521-source/texk/kpathsea/texmf.cnf	2015-10-01 11:49:11.703103900 -0400
@@ -229,7 +229,7 @@ TEXINPUTS.platex-ng = .;$TEXMF/tex/{upla
 % (e)up(La)TeX, and for upmpost
 TEXINPUTS.uplatex = .;$TEXMF/tex/{uplatex,platex,latex,generic,}//
 TEXINPUTS.uptex   = .;$TEXMF/tex/{uptex,ptex,plain,generic,}//
-TEXINPUTS.euptex  = .;$TEXMF/tex/{uptex,ptex,plain,generic}//
+TEXINPUTS.euptex  = .;$TEXMF/tex/{uptex,ptex,plain,generic,}//
 TEX.upmpost = euptex
 
 % pBibTeX bibliographies and style files.
