--- origsrc/CYGWIN-PATCHES/setup.hint	1969-12-31 19:00:00.000000000 -0500
+++ src/CYGWIN-PATCHES/setup.hint	2012-03-02 22:14:04.695956300 -0500
@@ -0,0 +1,7 @@
+category: Publishing
+requires: texlive-collection-basic
+sdesc: "TeX Live extra fonts"
+ldesc: "TeX Live is an easy way to get up and running with the TeX
+document production system.  It includes all the major TeX-related
+programs, macro packages, and fonts that are free software, including
+support for many languages around the world."
