--- origsrc/texlive-20110705-source/CYGWIN-PATCHES/libkpathsea-devel.hint	1969-12-31 19:00:00.000000000 -0500
+++ src/texlive-20110705-source/CYGWIN-PATCHES/libkpathsea-devel.hint	2012-04-26 21:40:15.071902500 -0400
@@ -0,0 +1,8 @@
+category: Publishing
+requires: libkpathsea6
+external-source: texlive
+sdesc: "TeX file and path search library (development)"
+ldesc: "TeX Live provides a comprehensive, cross-platform TeX system. It 
+includes all the major TeX-related programs, macro packages, and fonts 
+that are free software, including support for many languages around the 
+world."
--- origsrc/texlive-20110705-source/CYGWIN-PATCHES/libkpathsea6.hint	1969-12-31 19:00:00.000000000 -0500
+++ src/texlive-20110705-source/CYGWIN-PATCHES/libkpathsea6.hint	2012-04-26 21:39:11.821913800 -0400
@@ -0,0 +1,8 @@
+category: Publishing
+requires:
+external-source: texlive
+sdesc: "TeX file and path search library (runtime)"
+ldesc: "TeX Live provides a comprehensive, cross-platform TeX system. It 
+includes all the major TeX-related programs, macro packages, and fonts 
+that are free software, including support for many languages around the 
+world."
--- origsrc/texlive-20110705-source/CYGWIN-PATCHES/libptexenc-devel.hint	1969-12-31 19:00:00.000000000 -0500
+++ src/texlive-20110705-source/CYGWIN-PATCHES/libptexenc-devel.hint	2012-04-26 21:40:02.081026700 -0400
@@ -0,0 +1,8 @@
+category: Publishing
+requires: libkpathsea-devel libptexenc1
+external-source: texlive
+sdesc: "TeX Unicode encoding library (development)"
+ldesc: "TeX Live provides a comprehensive, cross-platform TeX system. It 
+includes all the major TeX-related programs, macro packages, and fonts 
+that are free software, including support for many languages around the 
+world."
--- origsrc/texlive-20110705-source/CYGWIN-PATCHES/libptexenc1.hint	1969-12-31 19:00:00.000000000 -0500
+++ src/texlive-20110705-source/CYGWIN-PATCHES/libptexenc1.hint	2012-04-26 21:39:49.546758900 -0400
@@ -0,0 +1,8 @@
+category: Publishing
+requires: libkpathsea6
+external-source: texlive
+sdesc: "TeX Unicode encoding library (runtime)"
+ldesc: "TeX Live provides a comprehensive, cross-platform TeX system. It 
+includes all the major TeX-related programs, macro packages, and fonts 
+that are free software, including support for many languages around the 
+world."
--- origsrc/texlive-20110705-source/CYGWIN-PATCHES/texlive.hint	1969-12-31 19:00:00.000000000 -0500
+++ src/texlive-20110705-source/CYGWIN-PATCHES/texlive.hint	2012-04-26 21:39:36.237226800 -0400
@@ -0,0 +1,7 @@
+category: Publishing
+requires: libfontconfig1 libfreetype6 libgcc1 libgd2 libgs9 libiconv2 libicu48 libkpathsea6 libpng14 libpoppler26 libptexenc1 libssp0 libstdc++6 libX11_6 libXaw7 libXmu6 libXpm4 libXt6 libzzip0.13 t1lib5 zlib0 texlive-collection-basic
+sdesc: "TeX Live binaries"
+ldesc: "TeX Live provides a comprehensive, cross-platform TeX system. It 
+includes all the major TeX-related programs, macro packages, and fonts 
+that are free software, including support for many languages around the 
+world."
