--- origsrc/CYGWIN-PATCHES/setup.hint	1969-12-31 19:00:00.000000000 -0500
+++ src/CYGWIN-PATCHES/setup.hint	2012-08-06 09:01:31.010721900 -0400
@@ -0,0 +1,10 @@
+category: Publishing
+requires: bash coreutils cygutils dialog ed fontconfig ghostscript grep perl sed texlive texlive-collection-documentation-base
+sdesc: "TeX Live basic packages"
+ldesc: "TeX Live is an easy way to get up and running with the TeX 
+document production system.  It includes all the major TeX-related 
+programs, macro packages, and fonts that are free software, including 
+support for many languages around the world."
+prev: 20120312-2
+curr: 20120628-1
+test: 20120628-2
\ No newline at end of file
--- origsrc/CYGWIN-PATCHES/texlive-collection-basic.postinstall	1969-12-31 19:00:00.000000000 -0500
+++ src/CYGWIN-PATCHES/texlive-collection-basic.postinstall	2012-08-06 09:03:08.513298700 -0400
@@ -0,0 +1,2 @@
+/usr/bin/texlive_fix_updmap.sh
+
--- origsrc/CYGWIN-PATCHES/texlive_fix_updmap.sh	1969-12-31 19:00:00.000000000 -0500
+++ src/CYGWIN-PATCHES/texlive_fix_updmap.sh	2012-08-06 09:00:54.730646800 -0400
@@ -0,0 +1,23 @@
+#! /bin/bash
+
+# Users who installed texlive-collection-*-20120628-1 when
+# /usr/share/texmf-site didn't exist found that the updmap-sys
+# commands in the postinstall scripts failed
+# (http://cygwin.com/ml/cygwin-apps/2012-08/msg00009.html).  This
+# script runs those commands.
+
+confdir=/etc/texmf/web2c
+if [ ! -f ${confdir}/.updmapfix ]
+then
+    mkdir -pv ${confdir}
+    touch ${confdir}/.updmapfix
+    rm -fv /etc/postinstall/texlive-collection-basic.sh.done
+    map_cmds=$(sed -n -e 's|^/usr/bin/updmap-sys --nohash --nomkmap \(--enable .*\)$|\1|p' /etc/postinstall/texlive-collection-*.sh.done)
+    if [ -n "$map_cmds" ]
+    then
+	/usr/bin/mktexlsr
+	/usr/bin/updmap-sys --nohash --nomkmap $map_cmds
+	/usr/bin/updmap-sys --nohash --syncwithtrees
+	/usr/bin/updmap-sys
+    fi
+fi
