--- origsrc/CYGWIN-PATCHES/setup.hint	1969-12-31 18:00:00.000000000 -0600
+++ src/CYGWIN-PATCHES/setup.hint	2012-02-03 04:24:58.832208700 -0600
@@ -0,0 +1,7 @@
+category: Publishing
+requires: texlive-collection-basic
+sdesc: "Documentation for texlive-collection-context"
+ldesc: "TeX Live is an easy way to get up and running with the TeX 
+document production system.  It includes all the major TeX-related 
+programs, macro packages, and fonts that are free software, including 
+support for many languages around the world."
