--- origsrc/bin/i386-cygwin/dvihp	2016-11-25 13:25:21.000000000 -0500
+++ src/bin/i386-cygwin/dvihp	2018-01-16 19:31:59.000000000 -0500
@@ -6,14 +6,11 @@
 
 : ${DVILJ=dvilj4}       # the dvilj variant to run
 : ${SPOOL=lpr}          # used to print an LJ file
-: ${TMPDIR=${TEMP-${TMP-/tmp}}} # for the dvicopy output
 
-tmpdir="${TMPDIR}"/dvihp$$
-(umask 077; mkdir "$tmpdir") \
+tmpdir=`mktemp -d` \
   || { echo "cannot create directory \`$tmpdir'."; exit 1; }
 
 trap '
-  cd "${TMPDIR}"
   test -d "$tmpdir" && { rm -f "$tmpdir"/*; rmdir "$tmpdir"; }
   exit 0
 ' 0 1 2 3 6 7 13 15
--- origsrc/bin/x86_64-cygwin/dvihp	2016-11-25 13:26:40.000000000 -0500
+++ src/bin/x86_64-cygwin/dvihp	2018-01-16 19:31:59.000000000 -0500
@@ -6,14 +6,11 @@
 
 : ${DVILJ=dvilj4}       # the dvilj variant to run
 : ${SPOOL=lpr}          # used to print an LJ file
-: ${TMPDIR=${TEMP-${TMP-/tmp}}} # for the dvicopy output
 
-tmpdir="${TMPDIR}"/dvihp$$
-(umask 077; mkdir "$tmpdir") \
+tmpdir=`mktemp -d` \
   || { echo "cannot create directory \`$tmpdir'."; exit 1; }
 
 trap '
-  cd "${TMPDIR}"
   test -d "$tmpdir" && { rm -f "$tmpdir"/*; rmdir "$tmpdir"; }
   exit 0
 ' 0 1 2 3 6 7 13 15
