--- origsrc/clisp-2.49.92/src/gllib/unistd.in.h	2018-02-11 15:33:24.000000000 -0500
+++ src/clisp-2.49.92/src/gllib/unistd.in.h	2018-02-21 13:08:11.857747500 -0500
@@ -400,13 +400,6 @@ _GL_WARN_ON_USE (dup3, "dup3 is unportab
 
 
 #if @GNULIB_ENVIRON@
-# if defined __CYGWIN__
-/* The 'environ' variable is defined in a DLL. Therefore its declaration needs
-   the '__declspec(dllimport)' attribute, but the system's <unistd.h> lacks it.
-   This leads to a link error on 64-bit Cygwin when the option
-   -Wl,--disable-auto-import is in use.  */
-_GL_EXTERN_C __declspec(dllimport) char **environ;
-# endif
 # if !@HAVE_DECL_ENVIRON@
 /* Set of environment variables and values.  An array of strings of the form
    "VARIABLE=VALUE", terminated with a NULL.  */
