diff -r 87cf29627c28 src/pathname.d
--- a/a/src/pathname.d	Mon Feb 19 19:46:09 2018 -0500
+++ b/b/src/pathname.d	Fri Feb 23 10:39:53 2018 -0500
@@ -138,40 +138,8 @@
           #ifdef HAVE_READLINK
             /* read symbolic link: */
             to_ptr[-1]=0; /* replace '/' with 0 */
-            #ifdef UNIX_CYGWIN
-            /* readlink() does not work right on NFS mounted directories
-             (it returns -1,ENOENT or -1,EIO).
-             So check for a directory first. */
-            var struct stat statbuf;
-            if (lstat(resolved_path,&statbuf) < 0) {
-              /* We know that /dev/fd on Cygwin is a symlink to /proc/self/fd,
-                 but the lstat() function does not know it: it returns -1,ENOENT.
-                 Override this behaviour. */
-              if (asciz_equal(resolved_path,"/dev/fd")) {
-                statbuf.st_mode = S_IFLNK | (S_IRUSR | S_IWUSR | S_IXUSR);
-              } else {
-                /* Error. */
-                return NULL;
-              }
-            }
-            if (S_ISDIR(statbuf.st_mode)) {
-              /* directory, not a symbolic link */
-              to_ptr[-1] = '/'; /* insert the '/' again */
-            } else if (!S_ISLNK(statbuf.st_mode)) {
-              /* something else, but not a directory or symbolic link. */
-              errno = ENOTDIR;
-              return NULL;
-            } else
-            #endif
               {
                 var int linklen;
-                #ifdef UNIX_CYGWIN
-                /* We know that /dev/fd on Cygwin is a symlink to /proc/self/fd,
-                   but the readlink() function does not know it. */
-                if (asciz_equal(resolved_path,"/dev/fd")) {
-                  memcpy(mypath,"/proc/self/fd",13); linklen = 13;
-                } else
-                #endif
                 {
                   linklen = readlink(resolved_path,mypath,sizeof(mypath)-1);
                 }
@@ -210,11 +178,7 @@
                   }
                   to_ptr = from_ptr;
                 } else {
-                  #if defined(UNIX_CYGWIN)
-                  if ((errno == EINVAL) || (errno == EACCES))
-                  #else
                   if (errno == EINVAL)
-                  #endif
                     /* no symbolic link */
                     to_ptr[-1] = '/'; /* insert the '/' again */
                   else
