--- origsrc/texlive-20170520-source/texk/web2c/luatexdir/image/pdftoepdf.w	2016-11-25 13:09:14.000000000 -0500
+++ src/texlive-20170520-source/texk/web2c/luatexdir/image/pdftoepdf.w	2017-06-24 18:22:36.639803800 -0400
@@ -71,7 +71,7 @@ static char *get_file_checksum(const cha
         ck = (char *) malloc(PDF_CHECKSUM_SIZE);
         if (ck == NULL)
             formatted_error("pdf inclusion","out of memory while processing '%s'", a);
-        snprintf(ck, PDF_CHECKSUM_SIZE, "%" PRIu64 "_%" PRIu64, (uint64_t) size,(uint64_t) mtime);
+        snprintf(ck, PDF_CHECKSUM_SIZE, "%"@= @>PRIu64 "_%"@= @>PRIu64, (uint64_t) size,(uint64_t) mtime);
    } else {
         switch (fe) {
             case FE_FAIL:
