--- origsrc/asymptote/glrender.cc	2018-10-06 11:06:40.000000000 -0400
+++ src/asymptote/glrender.cc	2018-10-06 11:09:56.159753200 -0400
@@ -4,10 +4,6 @@
  * Render 3D Bezier paths and surfaces.
  *****/
 
-#ifdef __CYGWIN__
-#define _POSIX_C_SOURCE 200809L
-#endif
-
 #include <stdlib.h>
 #include <fstream>
 #include <cstring>
--- origsrc/asymptote/main.cc	2018-10-06 11:06:40.000000000 -0400
+++ src/asymptote/main.cc	2018-10-06 11:09:56.163740700 -0400
@@ -19,10 +19,6 @@
 *
 *************/
 
-#ifdef __CYGWIN__
-#define _POSIX_C_SOURCE 200809L
-#endif
-
 #include <iostream>
 #include <cstdlib>
 #include <cerrno>
